examples/trading/api/prisma/migrations/20240122110930_init/migration.sql

```diff
@@ -0,0 +1,47 @@
+-- CreateTable
+CREATE TABLE "Locked" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "objectId" TEXT NOT NULL,
+    "keyId" TEXT,
+    "creator" TEXT,
+    "deleted" BOOLEAN NOT NULL DEFAULT false
+);
+
+-- CreateTable
+CREATE TABLE "Escrow" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "objectId" TEXT NOT NULL,
+    "sender" TEXT,
+    "recipient" TEXT,
+    "keyId" TEXT,
+    "swapped" BOOLEAN NOT NULL DEFAULT false,
+    "cancelled" BOOLEAN NOT NULL DEFAULT false
+);
+
+-- CreateTable
+CREATE TABLE "Cursor" (
+    "id" TEXT NOT NULL PRIMARY KEY,
+    "eventSeq" TEXT NOT NULL,
+    "txDigest" TEXT NOT NULL
+);
+
+-- CreateIndex
+CREATE UNIQUE INDEX "Locked_objectId_key" ON "Locked"("objectId");
+
+-- CreateIndex
+CREATE INDEX "Locked_deleted_idx" ON "Locked"("deleted");
+
+-- CreateIndex
+CREATE UNIQUE INDEX "Escrow_objectId_key" ON "Escrow"("objectId");
+
+-- CreateIndex
+CREATE INDEX "Escrow_cancelled_idx" ON "Escrow"("cancelled");
+
+-- CreateIndex
+CREATE INDEX "Escrow_recipient_idx" ON "Escrow"("recipient");
+
+-- CreateIndex
+CREATE INDEX "Escrow_swapped_idx" ON "Escrow"("swapped");
+
+-- CreateIndex
+CREATE INDEX "Escrow_sender_idx" ON "Escrow"("sender");
```

Comment:
This seems like a lot of indices -- potentially more than we need. I'd be open to us only adding indices as we need them (e.g. in the PR that also includes their use), but here are the indices I would have thought we needed:

- `Locked` on `keyId` -- to get information about the object that someone wants to trade their object for.
- `Locked` on `creator` -- to allow someone to view the objects they have locked and are therefore eligible to trade.
- `Escrow` on `recipient` -- to show the objects that someone is offering to trade with you.
- `Escrow` on `sender` -- to show the objects that you are offering to trade with someone.

And that should be it?

Reply:
Thanks, indeed I agree on simplifying here. The cancelled / swapped ones could be too much, depending on the distribution. If there are 90% cancelled, 10% non-cancelled, it would speed up things by a lot. But if there's a 50/50 distribution, then it would make no sense.

We could also simplify the schema and remove cancelled/swapped ones completely from the API.

The reason I build it that way is to accomplish consistency no matter which order these events are processed with (so it can execute all of the events in any order, asc or desc, without ending up in an invalid state). That was essentially a must when I didn't know the existence of `MoveEventModule` filter 😆